### PR TITLE
Update Structured Data Capture Library codelab

### PR DESCRIPTION
**Pull request by jingtang10**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

**Description**
- Remove some commented out code for user to copy+paste from codelab directly
- Use coroutine for extraction
- Fix issue where there isn't a questionnaire json string variable in the activity
- Add menu

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Documentation

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: codelabs/datacapture/app/src/main/java/com/example/foodallergy/MainActivity.kt [hunk 1]
```diff
@@ -14,33 +14,50 @@
...
+
+  var questionnaireJsonString :String? = null
+
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     setContentView(R.layout.activity_main)
 
-    // 4.1 Create the QuestionnaireFragment
-    // val fragment = QuestionnaireFragment()
+    // Replace with code from the codelab to configure a questionnaire fragment.
```
Comment (omarismail94) [inline codelabs/datacapture/app/src/main/java/com/example/foodallergy/MainActivity.kt:32]: Do we want to add the step number before the text here? So it becomes:

`4.1 Replace with code from the codelab to configure a questionnaire fragment.`

File: codelabs/datacapture/app/src/main/java/com/example/foodallergy/MainActivity.kt [hunk 2]
```diff
@@ -14,33 +14,50 @@
...
 
-    // 4.4 Add the questionnaire to the activity
-    // supportFragmentManager.commit {
-    //     add(R.id.fragment_container_view, fragment)
-    // }
+  private fun getStringFromAssets(fileName: String): String {
+    return assets.open(fileName).bufferedReader().use { it.readText() }
   }
-}
+}
```
Comment (omarismail94) [inline codelabs/datacapture/app/src/main/java/com/example/foodallergy/MainActivity.kt:63]: Did you run `spotlessApply`? usually, there is a newline at the end

File: codelabs/datacapture/app/src/main/java/com/example/foodallergy/MainActivity.kt (hunk 1 above)
Comment (omarismail94) [reply codelabs/datacapture/app/src/main/java/com/example/foodallergy/MainActivity.kt:32]: here and every where else with this text

File: codelabs/datacapture/app/src/main/java/com/example/foodallergy/MainActivity.kt (hunk 2 above)
Comment (jingtang10) [reply codelabs/datacapture/app/src/main/java/com/example/foodallergy/MainActivity.kt]: done

**Comment by jingtang10** [comment]
thanks omar - comments addressed.

**Review by omarismail94 — APPROVED** [review]
Nice change! 🎉

File: codelabs/datacapture/app/build.gradle.kts [hunk 3]
```diff
@@ -37,11 +37,6 @@ dependencies {
     implementation("com.google.android.material:material:1.4.0")
     implementation("androidx.constraintlayout:constraintlayout:2.1.2")
 
-    // 1.1 Add the Android FHIR SDK's SDC library
-    // implementation("com.google.android.fhir:data-capture:0.1.0-beta03")
-    // 1.2 Add the AndroidX Fragment library
-    // implementation("androidx.fragment:fragment-ktx:1.3.1")
-
```
Comment (williamito) [inline codelabs/datacapture/app/build.gradle.kts:44]: Should we leave a comment as a pointer for where to add the SDC Library dependencies?

**Pull request merged**